purchase_allowed_product/views/purchase_order_view.xml

```diff
@@ -22,7 +22,11 @@
             >
                 <attribute name="context">{
                     'restrict_supplier_id': parent.partner_id,
-                    'use_only_supplied_product': parent.use_only_supplied_product
+                    'use_only_supplied_product': parent.use_only_supplied_product,
+                    'partner_id':parent.partner_id,
```

Comment:
Which module adds this context keys?

Reply:
restrict_supplier_id and use_only_supplied_product are added by purchase _allowed_product
                    
partner_id, quantity, uom, and company_id are added by puchase (purchase_order_form view)

Reply:
OK, this is a "temporary patch", but what do you think about my general comment?

Reply:
If a dependecy towards base_view_inheritance_extension is set, wouldn't it be enough to add the context keys this way?
                <attribute name="context" operation="update">
                    {'restrict_supplier_id': parent.partner_id,
                    'use_only_supplied_product': parent.use_only_supplied_product}
                </attribute>

Why would overwriting fields_view_get method be necessary?

Reply:
They are 2 different techniques to choose between, not both

Reply:
OK, thank you.